### PR TITLE
Modify text :-)

### PR DESCRIPTION
**Pull request by kosayoda**
This PR implements fun commands to modify given text.
A utility function was implemented to replace multiple words in a string.

### CODE REVIEW

**Comment by kosayoda** [comment]
Missed a typehint :man_facepalming:, will batch push with review changes.

File: bot/seasons/evergreen/fun.py [hunk 1]
```diff
@@ -27,6 +40,48 @@ def __init__(self, bot):
...
+
+    @uwu_group.command(name="convert")
+    async def convert_uwu(self, ctx: Context, *, text: str) -> None:
+        """Converts a given `text` into it's uwu equivalent."""
+        text = await Fun.text_to_message(ctx, text)
+        converted = utils.replace_many(text, UWU_WORDS, ignore_case=True, match_case=True)
+        await ctx.send(f">>> {converted}")
+
+    @commands.group(name="randomcase", aliases=("rcase", "randomcaps", "rcaps",), invoke_without_command=True)
+    async def randomcase_group(self, ctx: Context, *, text: str) -> None:
```
Comment (MarkKoz) [inline bot/seasons/evergreen/fun.py]: I don't see the point in having a group that only has 1 subcommand.

File: bot/seasons/evergreen/fun.py [hunk 2]
```diff
@@ -27,6 +40,48 @@ def __init__(self, bot):
             output += getattr(Emojis, terning, '')
         await ctx.send(output)
 
+    @commands.group(name="uwu", aliases=("uwuwize", "uwuify",), invoke_without_command=True)
+    async def uwu_group(self, ctx: Context, *, text: str) -> None:
```
Comment (MarkKoz) [inline bot/seasons/evergreen/fun.py]: I don't see the point in having a group that only has 1 subcommand.

File: bot/seasons/evergreen/fun.py [hunk 3]
```diff
@@ -27,6 +40,48 @@ def __init__(self, bot):
...
+
+    @staticmethod
+    async def text_to_message(ctx: Context, text: str) -> str:
+        """
+        Attempts to convert a given `text` to a discord Message, then return the contents.
+
+        Returns `text` if the conversion fails.
+        """
+        try:
+            message = await MessageConverter().convert(ctx, text)
```
Comment (MarkKoz) [inline bot/seasons/evergreen/fun.py]: Why is this conversion needed? How would the contents after the conversion differ from the input and why do we care?

**Comment by MarkKoz** [comment]
Regarding the uwu text, it's a simple approach but it works. However, it got me interested in coming up with a more thorough approach to it, like [mapping graphemes to phonemes](https://github.com/dmort27/epitran#using-the-epitranvector-module) and determining replacements for the graphemes based on the phoneme they map to. This would allow for more thorough "uwuisation" of words while maintaining accuracy since it'd rely on the underlying phonemes rather than on the deep orthography of English and its irregular spelling.

File: bot/seasons/evergreen/fun.py (hunk 3 above)
Comment (sco1) [reply bot/seasons/evergreen/fun.py]: What was the answer to this question?

Comment (sco1) [reply bot/seasons/evergreen/fun.py]: Never mind, reading the docstring would help 💯

Comment (kosayoda) [reply bot/seasons/evergreen/fun.py]: Whoops, answered it elsewhere. It's so the commands work with a valid Discord message ID or link as an argument, if the conversion suceeds the text from the linked message gets converted.

**Pull request merged**